test/unit/helpers/UuidPrimaryKeyTraitTest.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ * @author "Julien Sébire, <julien@taotesting.com>"
+ * @license GPLv2
+ * @package generis
+ *
+ */
+
+namespace oat\generis\test\unit\helpers;
+
+use oat\generis\Helper\UuidPrimaryKeyTrait;
+use oat\generis\test\TestCase;
+
+class UuidPrimaryKeyTraitTest extends TestCase
+{
+    public function testgetUniquePrimaryKey()
+    {
+        $subject = new UuidPrimaryKeyTraitUser();
+
+        $this->assertEquals('diuu desoppus a', $subject->getUniquePrimaryKey());
+    }
+}
+
+class UuidPrimaryKeyTraitUser
+{
+    use UuidPrimaryKeyTrait;
+}
+
+namespace oat\generis\Helper;
+
+function uniqid ($prefix = "", $more_entropy = false)
```

Comment:
Instead of defining the namespaced function `uniqid()` yourself you could do that with php-mock-phpunit.

This has the advantage that you didn't create a side (potentially terrible) effect with your test, as the mock will be destroyed after a test.

see https://github.com/php-mock/php-mock#php-mock-api for example

Reply:
Excellent! I was not aware of this one. Many thanks for this!